### ISSUE TITLE
LoadError: cannot load such file -- rexml/document (runtime)

### ISSUE THREAD

**Issue by BastienDurel** [mastodon/mastodon#17039]
### Steps to reproduce the problem

after upgrading my OS to ruby 3, and mastodon to v3.4.3, I had an error during migration :

```
mastodon:~/live$ RAILS_ENV=production bundle exec rails db:migrate
rails aborted!
LoadError: cannot load such file -- rexml/document
/home/mastodon/live/vendor/bundle/ruby/3.0.0/gems/bootsnap-1.6.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:34:in `require'
[...]
```

to make it work, I had to add `rexml` as an explicit dependancy in the Gemfile (not only the Gemfile.lock)

```
Subject: [PATCH] upgrade deps for ruby 3

---
 Gemfile      | 1 +
 Gemfile.lock | 1 +
 2 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/Gemfile b/Gemfile
index dfcac636d..a9bb196c5 100644
--- a/Gemfile
+++ b/Gemfile
@@ -95,6 +95,7 @@ gem 'webpacker', '~> 5.4'
 gem 'webpush', '~> 0.3'
 gem 'bigdecimal'
 gem 'etc'
+gem 'rexml', '>= 3.2.5'
 gem 'webauthn', '~> 3.0.0.alpha1'
 
 gem 'json-ld'
diff --git a/Gemfile.lock b/Gemfile.lock
index 4018bb1e7..b0bf08efc 100644
--- a/Gemfile.lock
+++ b/Gemfile.lock
@@ -775,6 +775,7 @@ DEPENDENCIES
   redis (~> 4.2)
   redis-namespace (~> 1.8)
   resolv (~> 0.1.0)
+  rexml (>= 3.2.5)
   rqrcode (~> 2.0)
   rspec-rails (~> 5.0)
   rspec-sidekiq (~> 3.1)

```

### Expected behaviour

rails db:migrate misses rexml/document

### Actual behaviour

rails db:migrate runs without error

### Specifications

I'm using the v3.4.3 tag (with a small pach of my own to add `bigdecimal` & `etc` gems), under alpine linux

**Comment by ClearlyClaire** [comment]
I'm surprised I haven't experienced this, as I've been running Mastodon with Ruby 3 for a while. This should have been fixed in `main` with #16982 though.

**Comment by BastienDurel** [comment]
Maybe it's musl-specific ?
If it's already fixed, great :)
